docs/introduction.md

```diff
@@ -2,69 +2,146 @@
+ 
+- A Kubevela application is composed of various supported components or
+  [traits](https://github.com/oam-dev/spec/blob/master/introduction.md) and their
+  schemas are determined by the platform builders. New capabilities can be added
+  to Kubevela through a CRD registry mechanism. 
+
+- KubeVela provides a rich set of tools to help platform builders to abstract
+  the Kubernetes resource and CRD APIs.
+  For example, a [CUELang](https://github.com/cuelang/cue) based templating tool
+  is used to easily build the contract between the user-facing schemas and the
+  underline Kubernetes Objects. KubeVela provides a set of built-in CUE templates
+  for platform builders to start with.
+
+- Schema changes take effect immediately. Neither recompiliation nor redeployment
+  of KubeVela is required. This makes the process of delivering new capabilities
+  using Kubevela extremely simple.
+  
+With Kubevela, platform builders now have indefinite flexibilities in designing
+and implementing new capabilities without worrying about what and how to expose
+the new capabilities to the end users.
+
+The ultimate beneficiaries are the application developers. Instead of managing
+a handful Kubernetes YAML files, only a simple docker-compose style **appfile**
+is needed to manage an application in Kubevela.
+
+
+## Comparisons
+
+### Platform-as-a-Service (PaaS) 
+
+The typical examples are Heroku and Cloud Foundry. They also provide full
+application management capabilities and aim to improve developer experience
+and efficiency. KubeVela shares the same goal but its built-in features are
+much lighter and easier to maintain compared to most of the existing PaaS offerings.
+Kubevela core components are nothing but a set of Kubernetes controllers/plugins.
+
+The biggest difference lies in the extensibility. Most PaaS systems enforce
+constraints in the type of supported applications and the supported capabilities.
+They are either inextensible or create their own addon systems maintained by the
+user communities. In contrast, KubeVela is built on top of Kubernetes,
+and all the supported capabilities are implemented by Kubernetes CRD controllers.
+No additional addon system is introduced. A new capability can be installed in
+Kubevela at any time by simply registering the CRD in Kubevela.
+
+
+### Serverless platforms  
+
+Serverless platform such as AWS Lambda provides extraordinary user experience
+and agility to deploy serverless applications. However, those platforms impose
+even more constraints in extensibility. They are argurably "hard-coded" PaaS.
+
+Kubernetes based serverless platforms such as Knative, OpenFaaS can be easily
+integrated with Kubevela by registering themself as platform capabilities.
+Even for AWS Lambda, there is an success story to integrate it with Kubevela
+by the tools developed by Crossplane.
+
+### Platform agnostic developer tools
+
+The typical example is [Waypoint](https://github.com/hashicorp/waypoint). It is
```

Comment:
Suggestion:

```
The typical example is [Waypoint](https://github.com/hashicorp/waypoint). It is
a user facing tool which introduces a consistent workflow (i.e., build,
deploy, release) for developers to ship application on top of any platforms. KubeVela can be integrated into Waypoint 
like any other supported platforms. In this case, developers can use the Waypoint
workflow instead of the Kubevela CLI to manage applications.
```

Reply:
done